src/main/java/com/ericsson/ei/subscriptionhandler/InformSubscription.java

```diff
@@ -90,28 +91,42 @@
      * @param subscriptionJson
      */
     public void informSubscriber(String aggregatedObject, JsonNode subscriptionJson) {
-        String subscriptionName = subscriptionJson.get("subscriptionName").toString().replaceAll(REGEX, "");
-        LOGGER.debug("SubscriptionName : " + subscriptionName);
-        String notificationType = subscriptionJson.get("notificationType").toString().replaceAll(REGEX, "");
-        LOGGER.debug("NotificationType : " + notificationType);
-        String notificationMeta = subscriptionJson.get("notificationMeta").toString().replaceAll(REGEX, "");
-        LOGGER.debug("NotificationMeta : " + notificationMeta);
+        String subscriptionName = getSubscriptionField("subscriptionName", subscriptionJson);
+        String notificationType = getSubscriptionField("notificationType", subscriptionJson);
+        String notificationMeta = getSubscriptionField("notificationMeta", subscriptionJson);
+
+        String key = "";
+        String val = "";
         MultiValueMap<String, String> mapNotificationMessage = new LinkedMultiValueMap<>();
         ArrayNode arrNode = (ArrayNode) subscriptionJson.get("notificationMessageKeyValues");
         if (arrNode.isArray()) {
             for (final JsonNode objNode : arrNode) {
-                mapNotificationMessage.add(objNode.get("formkey").toString().replaceAll(REGEX, ""), jmespath
-                        .runRuleOnEvent(objNode.get("formvalue").toString().replaceAll(REGEX, ""), aggregatedObject)
-                        .toString().replaceAll(REGEX, ""));
+                if (objNode.get("formkey").toString().replaceAll(REGEX, "").equals("Authorization")) {
+                    key = "Authorization";
+                    val = objNode.get("formvalue").toString().replaceAll(REGEX, "");
+
+                } else {
+
+                    mapNotificationMessage.add(objNode.get("formkey").toString().replaceAll(REGEX, ""), jmespath
+                            .runRuleOnEvent(objNode.get("formvalue").toString().replaceAll(REGEX, ""), aggregatedObject)
+                            .toString().replaceAll(REGEX, ""));
+                }
             }
         }
         if (notificationType.trim().equals("REST_POST")) {
             LOGGER.debug("Notification through REST_POST");
             int result;
-            String headerContentMediaType = subscriptionJson.get("restPostBodyMediaType").toString()
-                    .replaceAll(REGEX, "");
+            String headerContentMediaType = subscriptionJson.get("restPostBodyMediaType").toString().replaceAll(REGEX,
+                    "");
             LOGGER.debug("headerContentMediaType : " + headerContentMediaType);
-            result = restTemplate.postDataMultiValue(notificationMeta, mapNotificationMessage, headerContentMediaType);
+            if (!key.isEmpty() && !val.isEmpty()) {
+                result = restTemplate.postDataMultiValue(notificationMeta, mapNotificationMessage,
```

Comment:
This is also not needed since handling of notificationMessageKeyValues should be done through only mapNotificationMessage like it was before.